### PR TITLE
TE-3.1: Base Hierarchical Route Installation - porting script changes from ATE to OTG location

### PR DESCRIPTION
**Pull request by stthummuri**
Ported changes from https://github.com/openconfig/featureprofiles/pull/850 to OTG location script 	https://github.com/openconfig/featureprofiles/blob/main/feature/gribi/otg_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go

### CODE REVIEW

**Comment by stthummuri** [comment]
@michaelore Can you please review and merge this PR

File: feature/gribi/otg_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go [hunk 1]
```diff
@@ -346,15 +348,20 @@ func testRecursiveIPv4Entry(t *testing.T, args *testArgs) {
 			t.Errorf("next-hop index is incorrect: got %v, want %v", got, want)
 		}
 		nh := gnmi.Get(t, args.dut, gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).Afts().NextHop(nhIndexInst).State())
-		if got, want := nh.GetIpAddress(), ateIndirectNH; got != want {
-			t.Errorf("next-hop is incorrect: got %v, want %v", got, want)
+		// for devices that return  the nexthop with resolving it recursively. For a->b->c the device returns c
+		if got := nh.GetIpAddress(); got != atePort2.IPv4 {
```
[the remark this answers is not in the document]
Comment (stthummuri) [reply feature/gribi/otg_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go]: Done.

File: feature/gribi/otg_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go [hunk 2]
```diff
@@ -313,15 +313,17 @@ func deleteRecursiveIPv4Entry(t *testing.T, args *testArgs) {
 	)
 }
 
-// testRecursiveIPv4Entry verifies recursive IPv4 Entry for 198.51.100.0/24 -> 203.0.113.1/32 -> 192.0.2.6.
+// testRecursiveIPv4Entry verifies recursive IPv4 Entry for 198.51.100.0/24 (a) -> 203.0.113.1/32 (b) -> 192.0.2.6 (c).
 // The IPv4 Entry is verified through AFT Telemetry and Traffic.
+// TODO: the below code checks entries for each level of the hierarchy statically. We need to create a helper function that does the check recursively.
```
[the remark this answers is not in the document]
Comment (stthummuri) [reply feature/gribi/otg_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go:318]: Done.

File: feature/gribi/otg_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go [hunk 3]
```diff
@@ -346,15 +348,20 @@ func testRecursiveIPv4Entry(t *testing.T, args *testArgs) {
 			t.Errorf("next-hop index is incorrect: got %v, want %v", got, want)
 		}
 		nh := gnmi.Get(t, args.dut, gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).Afts().NextHop(nhIndexInst).State())
-		if got, want := nh.GetIpAddress(), ateIndirectNH; got != want {
-			t.Errorf("next-hop is incorrect: got %v, want %v", got, want)
+		// for devices that return  the nexthop with resolving it recursively. For a->b->c the device returns c
+		if got := nh.GetIpAddress(); got != atePort2.IPv4 {
+			// for devices that return the nexthop without resolving it recursively. For a->b->c the device returns b
+			// note that b->c is validated in the next code block
```
[the remark this answers is not in the document]
Comment (stthummuri) [reply feature/gribi/otg_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go]: Done.

File: feature/gribi/otg_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go [hunk 4]
```diff
@@ -346,15 +348,20 @@ func testRecursiveIPv4Entry(t *testing.T, args *testArgs) {
 			t.Errorf("next-hop index is incorrect: got %v, want %v", got, want)
 		}
 		nh := gnmi.Get(t, args.dut, gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).Afts().NextHop(nhIndexInst).State())
-		if got, want := nh.GetIpAddress(), ateIndirectNH; got != want {
-			t.Errorf("next-hop is incorrect: got %v, want %v", got, want)
+		// for devices that return  the nexthop with resolving it recursively. For a->b->c the device returns c
```
[the remark this answers is not in the document]
Comment (stthummuri) [reply feature/gribi/otg_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go]: Done.

File: feature/gribi/otg_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (michaelore) [reply feature/gribi/otg_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go:318]: I was thinking we'd capitalize "the"->"The"

"TODO: The below code..."

**Review by michaelore — CHANGES_REQUESTED** [review]

**Comment by stthummuri** [comment]
Done --- 
I was thinking we'd capitalize "the"->"The"

"TODO: The below code..."

**Review by michaelore — APPROVED** [review]

**Pull request merged**